File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<pair<int, pair<int, int>>> pq;

        for (int i=0;i<points.size();i++) {
            pq.push({
                points[i][0] * points[i][0] + points[i][1] * points[i][1],
                {points[i][0], points[i][1]}
            });
        }

        for (int i=0;i<points.size()-k;i++) {
            pq.pop();
        }

        vector<vector<int>> res; 
        for (int i=0;i<k;i++) {
            pair<int, pair<int, int>> p = pq.top();
            res.push_back(vector<int>{p.second.first, p.second.second});
            pq.pop();
        }
        return res;
    }
};

File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.py
class Solution:
    def distance(self, point: List[int]) -> int:
        return point[0] ** 2 + point[1] ** 2

    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        arr = []
        for i, point in enumerate(points):
            arr.append((i, self.distance(point)))
    
        arr.sort(key=lambda x: x[1])

        res = []
        for i in range(k):
            res.append(points[arr[i][0]])
        
        return res
